code/parse/parselo.cpp

```diff
@@ -3466,6 +3466,11 @@ void stuff_matrix(matrix *mp)
 	stuff_vec3d(&mp->vec.rvec);
 	stuff_vec3d(&mp->vec.uvec);
 	stuff_vec3d(&mp->vec.fvec);
+
+	// Make sure this matrix uses normalized vectors.  All vectors in an orientation matrix must be normalized.
+	vm_vec_normalize(&mp->vec.rvec);
+	vm_vec_normalize(&mp->vec.uvec);
+	vm_vec_normalize(&mp->vec.fvec);
```

Comment:
For completeness, orthgonalization should probably be enforced too with `vm_orthogonalize_matrix`.

`vm_fix_matrix` is even more complete, handling normalization, orthgonalization, and any vectors which might be 0.

Reply:
Hmm.  This is a great idea, but will this cause any chaotic modification or the matrix itself?

What I'm worried about is repeated saves in FRED causing the matrix to drift.  I won't be able to run more tests for a few days.

Reply:
It shouldn't be particularly drastic unless the matrix was already unusual, and I don't have any hard data about it "drifting", but the operation is pretty mundane, and done in other contexts per frame, for instance, so it shouldn't be any worse or greater in magnitude than any drift an otherwise stationary object would receive every frame during gameplay, which in my experience is pretty much nil.

Reply:
Okay, fair enough.  Change pushed.